vendor/k8s.io/klog/v2/klog.go

```diff
@@ -1223,24 +1224,38 @@ func StartFlushDaemon(interval time.Duration) {
 // lockAndFlushAll is like flushAll but locks l.mu first.
 func (l *loggingT) lockAndFlushAll() {
 	l.mu.Lock()
-	l.flushAll()
+	files := l.flushAll()
 	l.mu.Unlock()
+	// Some environments are slow when syncing and holding the lock might cause contention.
+	l.syncAll(files)
 }
 
-// flushAll flushes all the logs and attempts to "sync" their data to disk.
+// flushAll flushes all the logs
 // l.mu is held.
-func (l *loggingT) flushAll() {
+func (l *loggingT) flushAll() []flushSyncWriter {
+	files := make([]flushSyncWriter, 0, severity.NumSeverity)
```

Comment:
how often is this called, is it an issue that this allocs a new slice on every call?

Reply:
We could mitigate by using an array owned by the caller instead of the slice.

Reply:
Now it's passed by value without allocating a slice.

Reply:
thanks!